src/pkg/packager/prepare.go

```diff
@@ -356,6 +384,31 @@ func (p *Packager) processUnstructuredImages(resource *unstructured.Unstructured
 	return matchedImages, maybeImages, nil
 }
 
+func findWhyResources(resources []*unstructured.Unstructured, whyImage, componentName, resourceName string, isChart bool) ([]string, error) {
+	foundWhyResources := []string{}
+	for _, resource := range resources {
+		bytes, err := resource.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		json := string(bytes)
+		resourceTypeKey := "manifest"
+		if isChart {
+			resourceTypeKey = "chart"
+		}
+
+		if strings.Contains(json, whyImage) {
+			yamlResource, err := yaml.Marshal(resource.Object)
+			if err != nil {
+				return nil, err
+			}
+			fmt.Printf("component: %s\n%s: %s\nresource:\n\n%s\n", componentName, resourceTypeKey, resourceName, string(yamlResource))
+			foundWhyResources = append(foundWhyResources, resourceName)
+		}
+	}
+	return foundWhyResources, nil
+}
```

Comment:
Looking at this function closer, we can simplify even further by only serializing the resource object once to YAML rather than serializing to JSON and then to YAML. This should be the last optimization we need to make to this logic 🤞 Thank you for your patience

```suggestion
func findWhyResources(resources []*unstructured.Unstructured, whyImage, componentName, resourceName string, isChart bool) ([]string, error) {
	foundWhyResources := []string{}
	for _, resource := range resources {
		bytes, err := yaml.Marshal(resource.Object)
		if err != nil {
			return nil, err
		}
		yaml := string(bytes)
		resourceTypeKey := "manifest"
		if isChart {
			resourceTypeKey = "chart"
		}

		if strings.Contains(yaml, whyImage) {
			fmt.Printf("component: %s\n%s: %s\nresource:\n\n%s\n", componentName, resourceTypeKey, resourceName, yaml)
			foundWhyResources = append(foundWhyResources, resourceName)
		}
	}
	return foundWhyResources, nil
}
```

Reply:
Oh I see what you did there ! No I appreciate the help, I'm learning in the process as well.